fs/dafsserver/dafsserver.cpp

```diff
@@ -2056,8 +2057,16 @@ class CRemoteIndexReadActivity : public CRemoteIndexBaseActivity
                 while (keyManager->lookup(true))
                 {
                     const byte *keyRow = keyManager->queryKeyBuffer();
-                    retSz = translator->translate(outBuilder, *this, keyRow);
-                    if (retSz)
+                    if (translator)
+                        retSz = translator->translate(outBuilder, *this, keyRow);
+                    else
+                    {
+                        outBuilder.ensureRow();
+                        retSz = keyManager->queryRowSize();
+                        memcpy(outBuilder.getSelf(), keyRow, retSz);
```

Comment:
@ghalliday - is this right way to copy index fields if no translation?

NB: before this fix and before HPCC-18484, it always used a translator - so this is secondary to the fix for this JIRA.

Reply:
This will have problems for variable length keys. I think it needs to be

    memcpy(outBuilder.ensureCapacity(retSz), keyRow, retSz);

You should be able to remove the ensureRow call as well.

Reply:
yes and actually I'd already changed/pushed it prior to your comment to:
```
                        retSz = keyManager->queryRowSize();
                        outBuilder.ensureCapacity(retSz, nullptr);
                        memcpy(outBuilder.getSelf(), keyRow, retSz);
```
but github has preserved old code with my/your comment it seems.
Think that's equivalent to what you have suggested.